portfolio/src/main/webapp/script.js

```diff
@@ -32,20 +32,40 @@ function addRandomGreeting() {
  * page.
  */
 function getComments() {
-  fetch('/data').then(response => response.json()).then((comments) => {
+  // Extract the number of comments selected from the dropdown.
+  let numCommentsSelect = document.getElementById('num-comments');
+  let numCommentsString = numCommentsSelect.options[numCommentsSelect.selectedIndex].value;
+
+  // Create fetch string and perform GET request.
+  let fetchString = '/data?num-comments=' + numCommentsString;
```

Comment:
Constructing URL parameters is one of those things you should leave to internal libraries as much as possible. At the very lease, you need to encode the field values so things like spaces and special characters get serialized properly (using `encodeURIComponent`).

Ideally, there's a library that does all the work for you. It looks like there might be a `URLSearchParams` you can use; I'm not sure if this is standard/core javascript, though. I leave it up to you to research that. Here's a starting point: https://stackoverflow.com/questions/111529/how-to-create-query-parameters-in-javascript

Reply:
Done. Let me know if the method I used isn't sufficient.